File: BNOI - Group C/Buses/Buses.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Buses
{
    static void Main()
    {
        string[] points = Console.ReadLine().Split(' ');
        int N = int.Parse(points[0]);
        List<int[]> Stops = new List<int[]>();
        for (int i = 0; i < N; i++)
        {
            string[] stop = Console.ReadLine().Split(' ');
            Stops.Add(new int[] { int.Parse(stop[0]), int.Parse(stop[1]), int.Parse(stop[2]) });
        }

        var routes = Stops.OrderBy(m => m[0]).ToList();

        Console.WriteLine(GetPrice(routes, int.Parse(points[1]), int.Parse(points[2])));
    }

    static int GetPrice(List<int[]> routes, int A, int B)
    {
        int price = 0;
        bool isRoute = false;
        foreach (var route in routes)
        {
            if (A >= route[0] && A < route[1])
            {
                isRoute = true;
                if (B >= route[0] && B < route[1]) return B - A;
                else price += TicketOrCard(route, A, false);
            }
            else if (B >= route[0] && B < route[1]) return price + TicketOrCard(route, B, true);
            else if (isRoute) price += route[2];
        }

        return price;
    }

    static int TicketOrCard(int[] route, int point, bool last)
    {
        int ticket = (last) ? point - route[0] : route[1] - point;
        return ticket < route[2] ? ticket : route[2];
    }
}

File: Bulgarian Competitions/BNOI - Group C/Buses/Buses.js
function solve(args) {
    var points = args[0].split(' ');
    var N = parseInt(points[0]);
    var routes = [];
    for (var i = 0; i < N; i++) {
        var stop = args[i + 1].split(' ');
        routes[i] = new Array(parseInt(stop[0]), parseInt(stop[1]), parseInt(stop[2]));
    }
    routes.sort(function (a, b) { return a[0] - b[0] });

    return getPrice(routes, parseInt(points[1]), parseInt(points[2]));

    function getPrice(routes, A, B) {
        var price = 0;
        var isRoute = false;
        for (var r in routes) {
            if (A >= routes[r][0] && A < routes[r][1]) {
                isRoute = true;
                if (B >= routes[r][0] && B < routes[r][1]) return B - A;
                else price += ticketOrCard(routes[r], A, false);
            }
            else if (B >= routes[r][0] && B < routes[r][1]) return price + ticketOrCard(routes[r], B, true);
            else if (isRoute) price += routes[r][2];
        }

        return price;
    }

    function ticketOrCard(r, point, last) {
        var ticket = (last) ? point - r[0] : r[1] - point;
        return ticket < r[2] ? ticket : r[2];
    }
}